### PR TITLE
update config server

### PR DESCRIPTION
**Pull request by kyessenov**

### CODE REVIEW

File: adapter/config/server/client_test.go [hunk 1]
```diff
@@ -9,8 +26,6 @@ import (
 	"time"
```
Comment (odeke-em) [inline adapter/config/server/client_test.go]: This space can go :)

File: adapter/config/server/handler.go [hunk 2]
```diff
@@ -76,30 +78,33 @@ func (api *API) GetConfig(request *restful.Request, response *restful.Response)
...
-	if err = request.ReadEntity(config); err != nil {
+	if err := request.ReadEntity(config); err != nil {
 		api.writeError(http.StatusBadRequest, err.Error(), response)
 		return
 	}
 
-	if err = config.ParseSpec(); err != nil {
+	byteSpec, err := json.Marshal(config.Content)
+	if err != nil {
+		api.writeError(http.StatusBadRequest, err.Error(), response)
```
Comment (odeke-em) [inline adapter/config/server/handler.go]: A missing return here?

File: adapter/config/server/apiserver_test.go [hunk 3]
```diff
@@ -32,25 +32,22 @@ import (
...
+		`"route":[{"tags":{"version":"v3"}}]}}`)
+
+	errItemExists  = &model.ItemAlreadyExistsError{Key: "name"}
+	errNotFound    = &model.ItemNotFoundError{Key: "name"}
 	errInvalidBody = errors.New("invalid character 'J' looking for beginning of value")
 	errInvalidSpec = errors.New("cannot parse proto message: json: " +
 		"cannot unmarshal string into Go value of type map[string]json.RawMessage")
-	errInvalidType = fmt.Errorf("unknown configuration type not-a-route-rule; use one of %v",
-		model.IstioConfigTypes.Types())
+	errInvalidType = fmt.Errorf("missing type")
```
Comment (odeke-em) [inline adapter/config/server/apiserver_test.go]: errInvalidType = errors.New("missing type")

File: adapter/config/server/apiserver.go [hunk 4]
```diff
@@ -12,7 +12,10 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package apiserver
```
Comment (rshriram) [inline adapter/config/server/apiserver.go]: Can you rename this folder to config/API so that we can have server.go and client.go underneath it?

**Comment by kyessenov** [comment]
closing the PR due to many merge conflicts and decision to move the code.
the branch `apiserver` will stay till when the new repo is ready

**Pull request closed without merging**